src/trixi.c

```diff
@@ -20,42 +20,34 @@ static int show_debug_output();
 // Auxiliary function to update JULIA_DEPOT_PATH environment variable
 static void update_depot_path(const char * project_directory, const char * depot_path);
 
-// Store function pointers to avoid overhead of `jl_eval_string`
-enum {
-  TRIXI_FTPR_INITIALIZE_SIMULATION,
-  TRIXI_FTPR_CALCULATE_DT,
-  TRIXI_FTPR_IS_FINISHED,
-  TRIXI_FTPR_STEP,
-  TRIXI_FTPR_FINALIZE_SIMULATION,
-  TRIXI_FTPR_NDIMS,
-  TRIXI_FTPR_NELEMENTS,
-  TRIXI_FTPR_NVARIABLES,
-  TRIXI_FTPR_LOAD_CELL_AVERAGES,
-
-  // The last one is for the array size
-  TRIXI_NUM_FPTRS
-};
-static void* trixi_function_pointers[TRIXI_NUM_FPTRS];
-
-// List of function names to obtain C function pointer from Julia
-// OBS! If any name is longer than 250 characters, adjust buffer sizes below
-static const char* trixi_function_pointer_names[] = {
-  [TRIXI_FTPR_INITIALIZE_SIMULATION] = "trixi_initialize_simulation_cfptr",
-  [TRIXI_FTPR_CALCULATE_DT]          = "trixi_calculate_dt_cfptr",
-  [TRIXI_FTPR_IS_FINISHED]           = "trixi_is_finished_cfptr",
-  [TRIXI_FTPR_STEP]                  = "trixi_step_cfptr",
-  [TRIXI_FTPR_FINALIZE_SIMULATION]   = "trixi_finalize_simulation_cfptr",
-  [TRIXI_FTPR_NDIMS]                 = "trixi_ndims_cfptr",
-  [TRIXI_FTPR_NELEMENTS]             = "trixi_nelements_cfptr",
-  [TRIXI_FTPR_NVARIABLES]            = "trixi_nvariables_cfptr",
-  [TRIXI_FTPR_LOAD_CELL_AVERAGES]    = "trixi_load_cell_averages_cfptr",
-};
-
```

Comment:
For once I moved the definition of the function pointer arrays to the header file, s.t. they become available in multiple c-files.
It allows to more clearly organize the C-API implementation. On the other hand the definition and declaration of the array get cumbersome (need for `extern`, ..., see below).

Reply:
The use of `extern` is canonical C and only required once in the header file, so that's ok imho.

However, what I wouldn't do is to put them in `trixi.h`. `trixi.h` is our externally visible include file and should only contain definitions of interest for an external user. OTOH, our function pointers are for internal use only. Thus, I would move the enum and the two global variable declarations to a new file, e.g., `function_pointers.h`, and the corresponding variable definitions to `function_pointers.c`.

Reply:
I also noted that we still prefix all file names with `trixi_`. While this is technically ok, we only really need `trixi.h` to have "trixi" in its name. Since it seems like we are now spreading the code to more files now, we could also consider to rename the files after their purpose, e.g., `api.c` (instead of `trixi.c`) and plain `version.c` or something similar/better

Reply:
Good points!

Maybe I revert the spreading and do a dedicated restructuring PR?

Reply:
Up to you. You could also try to finish this PR quickly as-is, and then create a new PR for cleanup.